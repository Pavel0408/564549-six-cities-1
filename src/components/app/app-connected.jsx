import connect from "react-redux/es/connect/connect";

import {ActionCreator} from "../../action-creator";
import {App} from "./app";
import {
  getActiveOffers,
  getCities,
  getCityName, getIsAuthorizationRequired,
  getOffersIsLoading,
  getOffersLoadError, getUser,
} from "../../reducer/selectors";
import {Operation} from "../../operation";
import {withScreenSwitch} from "../../hocs/with-screen-switch";
import {ScreenSwitch} from "../../hocs/screen-switch";


const mapStateToProps = (state) => {
  return {
    cityName: getCityName(state),
    offers: getActiveOffers(state),
    cities: getCities(state),
    isLoading: getOffersIsLoading(state),
    error: getOffersLoadError(state),
    isAuthorizationRequired: getIsAuthorizationRequired(state),
    user: getUser(state)
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    cityClickHandler: (changedCity) => {
      dispatch(ActionCreator.changeActiveCity(changedCity));
    },
    loadOffers: () => {
      dispatch(Operation.loadOffers());
    },
    isAuthorized: () => {
      dispatch(Operation.isAuthorized());
    },
    authorize: (evt) => {
      evt.preventDefault();
      const authorizationFormData = new FormData(evt.target);
      const email = authorizationFormData.get(`email`);
      const password = authorizationFormData.get(`password`);
      dispatch(Operation.authorize({
        email,
        password
      }));
    },
    signOut: (evt) => {
      evt.preventDefault();
      dispatch(ActionCreator.authorizationFailed());
    }
  };
};

const AppWithScreenSwitch = withScreenSwitch({
  Component: App,
  ScreenSwitch
});

export const AppConnected = connect(mapStateToProps, mapDispatchToProps)(AppWithScreenSwitch);

