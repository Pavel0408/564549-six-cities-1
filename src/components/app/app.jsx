import PropTypes from "prop-types";
import {PureComponent} from "react";
import {offersPropTypes} from "../../prop-types/offers-prop-types";

export class App extends PureComponent {
  constructor(props) {
    super(props);

    props.isAuthorized();
    this.props.loadOffers();
  }

  render() {
    return this.props.screenSwitch(this.props);
  }
}

App.propTypes = {
  offers: offersPropTypes,
  cityName: PropTypes.string.isRequired,
  cityClickHandler: PropTypes.func.isRequired,
  cities: PropTypes.arrayOf(PropTypes.string),
  isLoading: PropTypes.bool,
  error: PropTypes.object,
  loadOffers: PropTypes.func,
  isAuthorized: PropTypes.func,
  isAuthorizationRequired: PropTypes.bool,
  authorize: PropTypes.func,
  user: PropTypes.object,
  signOut: PropTypes.func,
  screenSwitch: PropTypes.func
};

