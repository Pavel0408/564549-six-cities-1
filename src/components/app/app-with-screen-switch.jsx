import {WithScreenSwitch} from "../../hocs/with-screen-switch";
import {OffersList} from "../offers-list/offers-list";
import React from "react";
import {App} from "./app";

export const AppWithScreenSwitch = <WithScreenSwitch render={(screenSwitch) =>{
  return <App {...screenSwitch} />;
}}/>;
