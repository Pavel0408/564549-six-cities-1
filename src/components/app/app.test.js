import React from 'react';
import renderer from 'react-test-renderer';
import {App} from './app.jsx';
import {BrowserRouter} from "react-router-dom";

const offersMock = [
  {
    name: `Beautiful & luxurious apartment at great location`,
    image: `img/apartment-01.jpg`,
    price: 120,
    rating: 5,
    isPremium: false,
    isFavorite: true,
    coordinates: [52.3909553943508, 4.85309666406198],
    city: `Amsterdam`,
    id: 0
  },
  {
    name: `Wood and stone place`,
    image: `img/room.jpg`,
    price: 80,
    rating: 10,
    isPremium: true,
    isFavorite: false,
    coordinates: [52.369553943508, 4.85309666406198],
    city: `Amsterdam`,
    id: 1
  }
];

const cityName = `Amsterdam`;

const cities = [
  `Paris`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`
];

const loadOffers = () => {
};

const isAuthorized = () => {
};

const onCityClick = () => {};

const screenSwitch = () => {
  return <div></div>;
};

describe(`testing App render`, () => {
  it(`App is render correctly`, () => {
    const tree = renderer.create(
        <BrowserRouter>
          <App
            offers={offersMock}
            cityName={cityName}
            cities={cities}
            onLoadOffers={loadOffers}
            onIsAuthorized={isAuthorized}
            screenSwitch={screenSwitch}
            onCityClick={onCityClick}
          />
        </BrowserRouter>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
