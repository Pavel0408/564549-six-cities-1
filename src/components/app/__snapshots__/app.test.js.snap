// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing App render App is render correctly 1`] = `<div />`;
